pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -189,7 +189,7 @@ func (m *kubeGenericRuntimeManager) generateContainerConfig(container *v1.Contai
 			return nil, err
 		}
 	} else if username != "" {
-		glog.Warningf("Non-root verification doesn't support non-numeric user (%s)", username)
```

Comment:
this shouldn't fail completely, right? it's only an issue if the effective security context disallows running as root.

I'd expect this instead:
```
// Verify RunAsNonRoot. Non-root verification only supports numeric user.
if err := verifyRunAsNonRoot(pod, container, uid, username); err != nil {
	return nil, err
}
```

with this change to verifyRunAsNonRoot:
```go
// verifyRunAsNonRoot verifies RunAsNonRoot.
func verifyRunAsNonRoot(pod *v1.Pod, container *v1.Container, uid *int64, username string) error {
	effectiveSc := securitycontext.DetermineEffectiveSecurityContext(pod, container)
	// If the option is not set, or if running as root is allowed, return nil.
	if effectiveSc == nil || effectiveSc.RunAsNonRoot == nil || !*effectiveSc.RunAsNonRoot {
		return nil
	}

	if effectiveSc.RunAsUser != nil {
		if *effectiveSc.RunAsUser == 0 {
			return fmt.Errorf("container's runAsUser breaks non-root policy")
		}
		return nil
	}

	if uid != nil && *uid == 0 {
		return fmt.Errorf("container has runAsNonRoot and image will run as root")
	}

	if uid == nil && len(username) > 0 {
		fmt.Errorf("container has runAsNonRoot and image has non-numeric user (%s), cannot verify user is non-root", username)
	}

	return nil
}
```

Reply:
Yes, exactly.  Only if the SC's non-root flag is set should this fail during the check.  And it doesn't look like we use the uid anywhere else in that method at a quick glance so we don't actually have to make the call to `getImageUser` (which calls the image service) if the SC doesn't require it.

Reply:
> And it doesn't look like we use the uid anywhere else in that method at a quick glance

@pweil- As far I can see, we use `uid` here: https://github.com/kubernetes/kubernetes/blob/2aeace402ac8dc88440eb5c0453b31dd28ca14f2/pkg/kubelet/kuberuntime/kuberuntime_container.go#L215

Reply:
Yeah, I missed that one :+1: